pkg/yttlibrary/overlay/array.go

```diff
@@ -4,13 +4,18 @@
 package overlay
 
 import (
+	"github.com/k14s/ytt/pkg/template"
 	"github.com/k14s/ytt/pkg/yamlmeta"
 )
 
 func (o Op) mergeArrayItem(
 	leftArray *yamlmeta.Array, newItem *yamlmeta.ArrayItem,
 	parentMatchChildDefaults MatchChildDefaultsAnnotation) error {
 
+	if !template.NewAnnotations(newItem).Has(AnnotationMatch) {
```

Comment:
It feels a little strange to me to see `NewAnnotations` here instead of `GetAnnotations`.
New is nice because it's properly typed, while Get simply returns an interface{}.

Does this feel any better to you?
`	if !newItem.GetAnnotations().(template.NodeAnnotations).Has(AnnotationMatch) {`

Reply:
if you look at impl of NewAnnotations it does exactly what you are suggesting (https://github.com/vmware-tanzu/carvel-ytt/blob/08ad8d2907e59743ee066f9724c1e3ec30060069/pkg/template/annotations.go#L27).

> I think what I'm really suggesting is to rename template.NewAnnotations. It's much more related to getting than new-ing

im not sure i have strong opinion on this (aside from if we do decide to a rename, it should be a separate commit), but i'll say that new-ing to me implies new-ing up an object (Annotations). it's an implementation detail of NewAnnotations that it's actually reading it off from an existing thing aka get-ing. im guessing based on naming it was only meant to indicate that this is a new object (all other constructors typically start with New).